### ISSUE TITLE
Data URIs getting relative URLs

### ISSUE THREAD

**Issue by Nettsentrisk** [cburgmer/inlineresources#13]
Debugging rasterizeHTML.js, I came across a problem with inlineresources'  handling of Data URIs in background images.

Through the function adjustPathsOfCssResources, all data URIs passed as background images will be given a relative URL based on the stylesheet they're within, which causes an incorrect external request for the image later on.

I experienced this when using SVGs as data URIs in stylesheets that were being run through rasterizeHTML.js.

**Comment by cburgmer** [comment]
Interesting find, thanks for issue. 

Can you give an example? The code loops through all background images (https://github.com/cburgmer/inlineresources/blob/master/src/inlineCss.js#L53) checking whether the given background image is a data URI (https://github.com/cburgmer/inlineresources/blob/master/src/util.js#L36), so technically should be taking care of this. There's also a test for this (https://github.com/cburgmer/inlineresources/blob/master/test/specs/InlineCssSpec.js#L109).

**Comment by Nettsentrisk** [comment]
It seems to be happening if the data URIs are like this in the CSS:
background-image:url("data:image/svg+xml,%EF%BB%BF%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20600%20600%22%3E%3Cpath%20d%3D%22M554.8%2C81.3V93.2H519.3V506.8h35.5v11.9H358V506.8h35.5V116.3h-2.6L294.2%2C518.7H213.6L100.1%2C114.4H97.5V506.8H133v11.9H45.2V506.8H80.7V93.2H45.2V81.3H219.4l85.8%2C321.8L382.6%2C81.3H554.8Z%22%20transform%3D%22translate(0)%22%20fill%3D%22%23fff%22%20%2F%3E%3C%2Fsvg%3E");

The code should be unquoting the string before it hits the isDataUri function, but it seems like something goes wrong here, because it's only the quoted data URI SVGs that this is happening to. I have unquoted values in the same CSS file and this error doesn't happen to them.

**Comment by Nettsentrisk** [comment]
Think I found the culprit, in extractCssUrl:
```javascript
exports.extractCssUrl = function (cssUrl) {
    var urlRegex = /^url\(([^\)]+)\)/,
        quotedUrl;

    if (!urlRegex.test(cssUrl)) {
        throw new Error("Invalid url");
    }

    quotedUrl = urlRegex.exec(cssUrl)[1];
    return cssSupport.unquoteString(trimCSSWhitespace(quotedUrl));
};
```
If you run `"url("data:image/svg+xml,%EF%BB%BF%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%20600%20600%22%3E%3Cpath%20d%3D%22M554.8%2C81.3V93.2H519.3V506.8h35.5v11.9H358V506.8h35.5V116.3h-2.6L294.2%2C518.7H213.6L100.1%2C114.4H97.5V506.8H133v11.9H45.2V506.8H80.7V93.2H45.2V81.3H219.4l85.8%2C321.8L382.6%2C81.3H554.8Z%22%20transform%3D%22translate(0)%22%20fill%3D%22%23fff%22%20%2F%3E%3C%2Fsvg%3E")"` in the .exec of the regex, you get `""data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg…oke-miterlimit%3D%2210%22%20style%3D%22transform%3A%20translate(-1px%2C0px"` out of it, which is incorrect.

The regex might not be the correct choice to test this type of value?

**Comment by cburgmer** [comment]
Thanks, you nailed the issue.
The simple parser stopped at the first closing bracket. The initial assumption was wrong in that brackets *do* turn up as values.
The committed fix now makes sure to honour quotes correctly, and should resolve your issue.

I agree that regex is a terrible choice, but it saves us a dependency on a proper css background parser.  I'm open though to pull one in if need be.

**Comment by sixinli** [comment]
woah just found this! this solves the hundreds of 404 in console issue for me!! 

thank you for the fix @cburgmer ! btw when will it be released?

**Comment by cburgmer** [comment]
Moved this up on my todo list :)

**Comment by cburgmer** [comment]
Released 0.4.0. Let me know if there are any issues. 
Thanks again for the bug report and the subsequent debugging.
